### ISSUE TITLE
Add a compilation with clang

### ISSUE THREAD

**Issue by EmilyBourne** [gyselax/gyselalibxx#29]
A clang compilation test in the CI would be beneficial for multiple reasons:
1. It would test compilation for Adastra
3. It would catch missing `KOKKOS_LAMBDA` errors (these are only warnings with a Nvidia compiler)

**Comment by EmilyBourne** [comment]
@tpadioleau What do you think is the best way to manage this? Should the main branch generate a new docker file?

**Comment by tpadioleau** [comment]
Yes I think we can do that, if we can use the latest llvm release it wouldbe great. We can use apt with https://apt.llvm.org
